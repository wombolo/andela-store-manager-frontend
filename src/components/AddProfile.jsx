import React, { Component } from 'react';
import {addNewProfile} from '../actions/profileActions';
import PropTypes from "prop-types";
import {connect} from "react-redux";

export class AddProfile extends Component {
  state={
    firstname: '',
    lastname: '',
    email: '',
    role: '',
    image: '',
    password: '',
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.addNewProfile(this.state);
  };

  handleChange = (event) => {
    this.setState({[event.target.id]: event.target.value});
  };

  render() {
    return (
      <div>
        <h1>Add New User</h1>

        <div className="row clearfix products-grid">
          <div className="col-10 m-0-auto m-t-5">
            <form onSubmit={this.handleSubmit}>

              <div className="row">

                <div className="col-3">
                  <figure className="image-holder">
                    <img alt={'image'} src={require("../assets/images/profile/team-7.jpg")} />
                    <input type={'file'} id={'image'}
                           onChange={ (e) => this.handleChange({
                             target:{
                               id:'image',
                               value: e.target.files[0],
                             }
                           })
                           }/>

                  </figure>
                </div>


                <div className="col-7">
                  <table border="0" width="100%" className="product-single-table table-striped">
                    <tbody>
                    <tr>
                      <td><label htmlFor="firstname">First Name</label></td>
                      <td><input id="firstname" type="text" required={true} onChange={this.handleChange}/></td>
                    </tr>

                    <tr>
                      <td><label htmlFor="lastname">Last Name</label></td>
                      <td><input id="lastname" type="text" required={true} onChange={this.handleChange}/></td>
                    </tr>

                    <tr>
                      <td><label htmlFor="email">Email</label></td>
                      <td><input id="email" type="email" required={true} onChange={this.handleChange}/></td>
                    </tr>

                    <tr>
                      <td><label>Role</label></td>
                      <td>
                        <select id={'role'} onChange={this.handleChange}>
                          <option>---Choose One---</option>
                          <option value='store_attendant'>Attendant</option>
                          <option value='admin'>Admin</option>
                        </select>
                      </td>
                    </tr>

                    <tr>
                      <td><label htmlFor="password">Password</label></td>
                      <td><input id="password" type="password" onChange={this.handleChange}/></td>
                    </tr>
                    <tr>
                      <td><label htmlFor="conf_password">Confirm Password</label></td>
                      <td><input id="conf_password" type="password" onChange={this.handleChange}/></td>
                    </tr>

                    <tr>
                      <td>&nbsp;</td>
                      <td><button type="submit" className="btn btn-primary">Save</button>
                        <button type="reset" className="btn btn-danger">Clear</button></td>
                    </tr>
                    </tbody>

                  </table>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    )
  }
}

AddProfile.propTypes = {
  addNewProfile: PropTypes.func.isRequired
};

export default connect(null, {
  addNewProfile,
})(AddProfile);
