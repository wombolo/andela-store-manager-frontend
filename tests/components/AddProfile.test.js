import 'babel-polyfill'
import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';

import { AddProfile } from '../../src/components/AddProfile.jsx';

let wrapper;

const props = {
  addNewProfile: () => null,
};


describe('AddProduct Component', () => {
  it('should match snapshot', () => {
    wrapper = shallow(<AddProfile {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Methods in Class', () => {

    beforeEach(() => {
      // store = createStore(combineReducers({}));
      wrapper = mount(<AddProfile {...props}/>);
    });

    afterEach(() => {
      wrapper.unmount();
    });

    it('triggers handleSubmit()', () => {
      // const image_file = wrapper.find('form').last();
      // image_file.simulate('submit', {preventDefault: () => {}});

      sinon.spy(wrapper.instance(), 'handleSubmit');
      wrapper.instance().handleSubmit({preventDefault: () => {}});
      expect(wrapper.instance().handleSubmit.calledOnce)
        .toEqual(true);
    });

    it('triggers handleChange() for input title', () => {
      sinon.spy(wrapper.instance(), 'handleChange');
      wrapper.instance().handleChange({target: { value: 'u' , id: 'title'} });
      expect(wrapper.instance().handleChange.calledOnce)
        .toEqual(true);
    });

    it('triggers handleChange() for image input', () => {
      const file = {lastModified: 1547454659333,
      lastModifiedDate: 'Mon Jan 14 2019 09:30:59 GMT+0100 (West Africa Standard Time) {}',
      name: "15158524_xl.jpg",
      size: 3734559,
      type: "image/jpeg",
      webkitRelativePath: ""};

      const image = wrapper.find('#image').last();
      image.simulate('change', { target: { files: [file] , id: 'image'} });
    });

  });
});
