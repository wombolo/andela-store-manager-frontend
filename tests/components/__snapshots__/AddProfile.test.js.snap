// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProduct Component should match snapshot 1`] = `
<div>
  <h1>
    Add New User
  </h1>
  <div
    className="row clearfix products-grid"
  >
    <div
      className="col-10 m-0-auto m-t-5"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="row"
        >
          <div
            className="col-3"
          >
            <figure
              className="image-holder"
            >
              <img
                alt="image"
                src="../assets/images/profile/team-7.jpg"
              />
              <input
                id="image"
                onChange={[Function]}
                type="file"
              />
            </figure>
          </div>
          <div
            className="col-7"
          >
            <table
              border="0"
              className="product-single-table table-striped"
              width="100%"
            >
              <tbody>
                <tr>
                  <td>
                    <label
                      htmlFor="firstname"
                    >
                      First Name
                    </label>
                  </td>
                  <td>
                    <input
                      id="firstname"
                      onChange={[Function]}
                      required={true}
                      type="text"
                    />
                  </td>
                </tr>
                <tr>
                  <td>
                    <label
                      htmlFor="lastname"
                    >
                      Last Name
                    </label>
                  </td>
                  <td>
                    <input
                      id="lastname"
                      onChange={[Function]}
                      required={true}
                      type="text"
                    />
                  </td>
                </tr>
                <tr>
                  <td>
                    <label
                      htmlFor="email"
                    >
                      Email
                    </label>
                  </td>
                  <td>
                    <input
                      id="email"
                      onChange={[Function]}
                      required={true}
                      type="email"
                    />
                  </td>
                </tr>
                <tr>
                  <td>
                    <label>
                      Role
                    </label>
                  </td>
                  <td>
                    <select
                      id="role"
                      onChange={[Function]}
                    >
                      <option>
                        ---Choose One---
                      </option>
                      <option
                        value="store_attendant"
                      >
                        Attendant
                      </option>
                      <option
                        value="admin"
                      >
                        Admin
                      </option>
                    </select>
                  </td>
                </tr>
                <tr>
                  <td>
                    <label
                      htmlFor="password"
                    >
                      Password
                    </label>
                  </td>
                  <td>
                    <input
                      id="password"
                      onChange={[Function]}
                      type="password"
                    />
                  </td>
                </tr>
                <tr>
                  <td>
                    <label
                      htmlFor="conf_password"
                    >
                      Confirm Password
                    </label>
                  </td>
                  <td>
                    <input
                      id="conf_password"
                      onChange={[Function]}
                      type="password"
                    />
                  </td>
                </tr>
                <tr>
                  <td>
                    Â 
                  </td>
                  <td>
                    <button
                      className="btn btn-primary"
                      type="submit"
                    >
                      Save
                    </button>
                    <button
                      className="btn btn-danger"
                      type="reset"
                    >
                      Clear
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
